create_geometry_3D.py

```diff
@@ -0,0 +1,258 @@
+import gmsh
+import sys
+import numpy as np
+# from points_input import input_points, mesh_size, dims, depth, save_file, gmsh_interface
+
+
+# define the points of the surface as a list of tuples
+input_points = [(0, 0, 0), (1, 0, 0), (1, 3, 0), (0, 3, 0), (-1, 1.5, 0)]
+# define the mesh size
+mesh_size = 2
+# define geometry dimension; input "3" for 3D to extrude the 2D surface, input "2" for 2D
+dims = 3
+# input depth of geometry if 3D
+depth = 2
+# set a name label for the surface
+name_label = "Soil Layer"
+# if True, saves mesh data to separate mdpa files
+save_file = True
+# if True, opens gmsh interface
+gmsh_interface = True
+# set a name for mesh output file
+mesh_output_name = "geometry"
+
+
+def create_point(coordinates, mesh_size):
+    """
+    creates points in gmsh
+    :param coordinates: gets points coordinates in order from user
+    :param mesh_size: gets mesh size from user
+    :return: -
+    """
+    x = coordinates[0]
+    y = coordinates[1]
+    z = coordinates[2]
+    lc = mesh_size
+    gmsh.model.geo.addPoint(x, y, z, lc)
+
+
+def create_line(point_ids):
+    """
+    creates lines in gmsh
+    :param point_ids: gets point tags in order
+    :return: -
+    """
+    point1 = point_ids[0]
+    point2 = point_ids[1]
+    gmsh.model.geo.addLine(point1, point2)
+
+
+def create_surface(line_ids, name_label):
+    """
+    creates curve and then surface in gmsh by using line tags
+    :param line_ids: gets line tags in order
+    :param name_label: surface name label from user input
+    :return: returns the surface tag and surface name label
+    """
+    gmsh.model.geo.addCurveLoop(line_ids, 1)
+    surfaces = gmsh.model.geo.addPlaneSurface([1], 1)
+    gmsh.model.setPhysicalName(2, surfaces, name_label)
+    return surfaces
+
+
+def create_volume(surface_id, depth):
+    """
+    creates volume by extruding 2D surface
+    :param surface_id: surface tag
+    :param depth: depth of 3D geometry
+    :return: -
+    """
+    gmsh.model.geo.extrude([(2, surface_id)], 0, 0, depth)
+
+
+def make_elements(point_coordinates, point_pairs, mesh_size, name_label):
```

Comment:
I would put everything from this function in make_geometry_2D, and then call make_geometry_2D from make_geometry_3D